### PR TITLE
Rename FiniteField -> finite_field

### PR DESCRIPTION
**Pull request by fingolfin**
Also add a backwards compatibility @alias

So that the examples I mentioned today work

### CODE REVIEW

**Review by lgoettgens — APPROVED** [review]
I hope you didn't do all of this by hand :)
Looks good, but I did not check explicitly, that you indeed catched all occurrences.

**Comment by lgoettgens** [comment]
Unfortunately, this seems to break Hecke

**Comment by fingolfin** [comment]
Of course I didn't do this by hand ;-) rather I used this: `git grep -l -w FiniteField | xargs perl -pi -e 's;\bFiniteField\b;finite_field;g'` and then manually added the `@alias`.

I'm surprised this breaks Hecke (as we have an `@alias` and `finite_field` is not mentioned in Hecke -- but I'll investigate!

**Comment by fingolfin** [comment]
Ah, Hecke accesses `Native.FiniteField`.

So either I also need an alias inside `Native`, or, perhaps better, we just leave `Native.FiniteField` unrenamed. Thoughts?

**Comment by lgoettgens** [comment]
> Ah, Hecke accesses `Native.FiniteField`.
> 
> So either I also need an alias inside `Native`, or, perhaps better, we just leave `Native.FiniteField` unrenamed. Thoughts?

My way to go: Add a temporary fix here (like you did), then adapt Hecke to use `Native.finite_field` and in one of the next breaking Nemo releases remove the fix.

**Comment by lgoettgens** [comment]
> Pushed a fix using `@alias` inside `Native`

I think `@alias` needs to get imported into `Native`.

File: src/Native.jl [hunk 1]
```diff
@@ -21,51 +21,53 @@ function GF(n::ZZRingElem; cached::Bool=true)
...
   return k, k(1)
 end
 
-GF(p::Integer, k::Int, s::VarName = :o; cached::Bool = true) = FiniteField(p, k, s, cached = cached)[1]
+GF(p::Integer, k::Int, s::VarName = :o; cached::Bool = true) = finite_field(p, k, s, cached = cached)[1]
 
-GF(p::ZZRingElem, k::Int, s::VarName = :o; cached::Bool = true) = FiniteField(p, k, s, cached = cached)[1]
+GF(p::ZZRingElem, k::Int, s::VarName = :o; cached::Bool = true) = finite_field(p, k, s, cached = cached)[1]
+
+@alias FiniteField finite_field # for compatibility with Hecke
```
Comment (fingolfin) [inline src/Native.jl:70]:
```suggestion
import ..Nemo: @alias
@alias FiniteField finite_field # for compatibility with Hecke
```

**Pull request merged**